llvm/lib/Target/AMDGPU/AMDGPURemoveIncompatibleFunctions.cpp

```diff
@@ -139,10 +139,10 @@ bool AMDGPURemoveIncompatibleFunctions::checkFunction(Function &F) {
   const GCNSubtarget *ST =
       static_cast<const GCNSubtarget *>(TM->getSubtargetImpl(F));
 
-  // Check the GPU isn't generic. Generic is used for testing only
-  // and we don't want this pass to interfere with it.
+  // Check the GPU isn't generic or generic-hsa. Generic is used for testing
+  // only and we don't want this pass to interfere with it.
   StringRef GPUName = ST->getCPU();
-  if (GPUName.empty() || GPUName.contains("generic"))
+  if (GPUName.empty() || GPUName.starts_with("generic"))
```

Comment:
Are we sure this is correct? The existing issue (as far as I can tell) is that we never really implemented `generic`, so it'd just end up as some specific. However, with generic targets, it appears that the pass would actually make sense / apply / have some use?

Reply:
No we have some tests using a generic target, and I want the pass to work on -generic targets as well (e.g. gfx9-generic), so I'm moving the check to allow stuff like gfx9-generic but not "generic" alone

Reply:
we should probably start refusing to codegen "generic" or whatever none ends up getting called